Automattic/jetpack
# AI Assistant: Add treatment for error when prompt is not understood

Fixes #30651

## Proposed changes:

* Adds a rule so that the AI responds with a specific token when the prompt is not understood
* Adds an initial buffer to check for that error token

This PR is focused more in the mechanism, not much in the UI, as it is being defined and will be improved soon.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Add an AI Assistant block
* Ask some nonsense like "fdeswjfdaskfj"
* Check that a notice appears asking for a clear input
* Generate some valid content
* Ask some nonsense again
* Check that the notice appears and the previous content is not overwritten

![2023-05-17_15-35-47](https://github.com/Automattic/jetpack/assets/8486249/7a9711b2-5e4b-4118-bf83-c0f66dc732b3)
![2023-05-17_15-35-30](https://github.com/Automattic/jetpack/assets/8486249/74f8b381-423c-401f-b180-3ae6574d0ec3)
